nuitka/BytecodeCaching.py

```diff
@@ -83,24 +84,26 @@ def getCachedImportedModulesNames(module_name, source_code):
     if data["module_name"] != module_name:
         return None
 
-    return [
-        (ModuleName(used_module_name), line_number)
-        for (used_module_name, line_number) in data["modules_used"]
-    ]
+    used_modules = list(data["modules_used"])
+    for module in used_modules:
+        module["module_name"] = ModuleName(module["module_name"])
+
+    return [ModuleUsageAttempt(**module) for module in used_modules]
 
 
 def writeImportedModulesNamesToCache(module_name, source_code, used_modules):
     cache_name = makeCacheName(module_name, source_code)
     cache_filename = _getCacheFilename(cache_name, "json")
 
+    used_modules = [module._asdict() for module in used_modules]
+    for module in used_modules:
+        module["source_ref"] = module["source_ref"].getLineNumber()
```

Comment:
The `source_ref` is not restored it seems. Probably needs similar treatment as the `ModuleName()`.

Reply:
it becomes restored in` _loadUncompiledModuleFromCache` `source_ref.atLineNumber(module.source_ref)`